File: shared/src/main/java/com/gcgenome/lims/data/Preprocessing.java
package com.gcgenome.lims.data;

import jsinterop.annotations.JsIgnore;
import jsinterop.annotations.JsOverlay;
import jsinterop.annotations.JsPackage;
import jsinterop.annotations.JsType;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@JsType(isNative=true, namespace= JsPackage.GLOBAL, name="Object")
@Setter(onMethod_= {@JsOverlay, @JsIgnore})
@Getter(onMethod_= {@JsOverlay, @JsIgnore})
@Accessors(fluent=true)
public final class Preprocessing {
    private String id;
    private String lastModifyBy;
    private String lastModifyAt;
    private String createdBy;
    private String createdAt;
    private Double concNa;
    private Double concInput;
    private Double libPrep;
    private Double libConcTape;
    private Double libConcQubit;
    private Double fragmentSize;
    private Double amount;
    private Double dilution;
    private Double volume;
    private Double libVolume;
    private Double bufferVolume;
    private String indexI7;
    private String sequenceI7;
    private String indexI5;
    private String sequenceI5;
    private String json;
}

File: worklist-api/src/main/kotlin/com/gcgenome/lims/data/Preprocessing.kt
package com.gcgenome.lims.data

data class Preprocessing(
    val worklist: String,
    val index: Int,
    val concNa: Double?,
    val concInput: Double?,
    val libPrep: String?,
    val libConcTape: Double?,
    val libConcQubit: Double?,
    val fragmentSize: Double?,
    val amount: Double?,
    val dilution: Double?,
    val volume: Double?,
    val libVolume: Double?,
    val bufferVolume: Double?,
    val indexI7: String?,
    val sequenceI7: String?,
    val indexI5: String?,
    val sequenceI5: String?
)

